libs/dmlf/tests/execution/promise_of_result.cpp

```diff
@@ -0,0 +1,96 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "dmlf/execution/execution_result.hpp"
+
+#include "gtest/gtest.h"
+
+#include <stdexcept>
+
+namespace {
+
+using fetch::dmlf::ExecutionResult;
+using PromiseOfResult = fetch::dmlf::ExecutionResult::PromiseOfResult;
+using State           = fetch::service::details::PromiseImplementation::State;
+using fetch::dmlf::ExecutionErrorMessage;
```

Comment:
Just a style thing. I usually keep `using foo = blah` aliases and `using foo::bar` include in two groups if possible